misc/userscripts/qutebrowser_viewsource

```diff
@@ -24,9 +24,15 @@
 # Caveat: Does not use authentication of any kind. Add it in if you want it to.
 #
 
+TERMINAL=${TERMINAL:-urxvt}
+EDITOR=${EDITOR:-vim}
+
 path=/tmp/qutebrowser_$(mktemp XXXXXXXX).html
+curl "$QUTE_URL" > "$path"
 
-curl "$QUTE_URL" > $path
-urxvt -e vim "$path"
+unset _command
+[[ $EDITOR =~ emacs|gedit|leafpad|atom ]] && _command="$EDITOR $path"
+[ -z "$_command" ] && _command="$TERMINAL -e \"vim $path\""
 
+eval "$_command"
```

Comment:
The whole way of doing it is very odd. I think it doesn't even work with terminal editors because the arguments after -e are used to start the process without any further parsing. This should be better:
```
if [[ $EDITOR =~ emacs|gedit|leafpad|atom ]]; then
    "$EDITOR" "$path"
else
    "$TERMINAL" -e "$EDITOR" "$path"
fi
```